crates/sui-indexer-alt/src/lib.rs

```diff
@@ -368,30 +355,71 @@ impl Indexer {
     }
 }
 
+impl Default for IndexerArgs {
+    fn default() -> Self {
+        Self {
+            first_checkpoint: None,
+            last_checkpoint: None,
+            pipeline: vec![],
+            metrics_address: "0.0.0.0:9184".parse().unwrap(),
+        }
+    }
+}
+
 pub async fn start_indexer(
+    db_args: DbArgs,
+    indexer_args: IndexerArgs,
     indexer_config: IndexerConfig,
-    db_config: DbConfig,
-    consistency_config: ConsistencyConfig,
     // If true, the indexer will bootstrap from genesis.
     // Otherwise it will skip the pipelines that rely on genesis data.
     // TODO: There is probably a better way to handle this.
     // For instance, we could also pass in dummy genesis data in the benchmark mode.
     with_genesis: bool,
 ) -> anyhow::Result<()> {
+    let IndexerConfig {
+        ingestion,
+        consistency,
+        pipeline:
+            PipelineConfig {
+                sum_coin_balances,
+                wal_coin_balances,
+                sum_obj_types,
+                wal_obj_types,
+                sum_displays,
+                sum_packages,
+                ev_emit_mod,
+                ev_struct_inst,
+                kv_checkpoints,
+                kv_epoch_ends,
+                kv_epoch_starts,
+                kv_feature_flags,
+                kv_objects,
+                kv_protocol_configs,
+                kv_transactions,
+                obj_versions,
+                tx_affected_addresses,
+                tx_affected_objects,
+                tx_balance_changes,
+                tx_calls,
+                tx_digests,
+                tx_kinds,
+            },
+    } = indexer_config;
```

Comment:
I love destructuring like this. I find myself using it more and more, although it can get verbose.